### ISSUE TITLE
bugs on resizing windows

### ISSUE THREAD

**Issue by engineer1109** [jherico/Vulkan#42]
terminate called after throwing an instance of 'vk::OutOfDateKHRError'
  what():  vk::Queue::presentKHR: ErrorOutOfDateKHR

Segmentation fault
Happens on resizing windows.
System : Linux Ubuntu-18.04.2

**Comment by engineer1109** [comment]
swapChain.queuePresent(submitOverlay ? semaphores.overlayComplete : semaphores.renderComplete)
is error

**Comment by engineer1109** [comment]
now becomes 50% chance.

terminate called after throwing an instance of 'vk::OutOfDateKHRError'
  what():  vk::Queue::presentKHR: ErrorOutOfDateKHR

**Comment by jherico** [comment]
Yeah, something about my resizing code isn't properly waiting.  I'll try to investigate this.

**Comment by engineer1109** [comment]
@jherico  use try catch may avoid it

**Comment by engineer1109** [comment]
@jherico  For the triangle sample, I have used this to fix
### Code
        try {
            swapChain.queuePresent(semaphores.renderComplete);
        } catch (std::exception e) {
            std::cout<<"error"<<std::endl;
            swapChain.create(size);
        }

**Comment by engineer1109** [comment]
@jherico  
Parts of the samples can successfully resize. Like texture3d and viewportarray.
Fix bugs with this in vulkanExampleBase.cpp
379 line

### cxx code
try {
        swapChain.queuePresent(submitOverlay ? semaphores.overlayComplete : semaphores.renderComplete);
    } catch (std::exception e) {
        swapChain.create(size);
    }

Still bugs exist with multiview and subpass .etc.
